### ISSUE TITLE
To incorporate simple informed and uninformed algorithms

### ISSUE THREAD

**Issue by Korra15** [codezonediitj/pydatastructs#447]
#### Description of the problem
Python provides us with an amazing and easy way to implement informed and uninformed search algorithms. These algorithms help programmers to easily help their agent find it's goal in the best way possible. I feel that these algorithms should be included along with the others.

Could you please assign this to me under SWOC 32.0

**Comment by czgdp1807** [comment]
Hi @Korra15 . The idea seems interesting. Could you please provide references (like research papers, blog posts, university lecture notes) so that we can discuss more?

**Comment by Korra15** [comment]
Yeah sure @czgdp1807, following is the link of the pdf that explains mre about these algorithms.
[http://i-rep.emu.edu.tr:8080/jspui/bitstream/11129/4714/1/algasikhaled.pdf](url)

**Comment by czgdp1807** [comment]
I see. So, these are from artificial intelligence domain. Well, in general all the algorithms in paper are very well applicable on implicit graphs. More like, you know the current node and you also know a function to generate the neighbours of the current node. So, only these two pieces of information are used in these algorithms. Hence, I would recommend, first adding an implicit graph data structure (see #224) and then defining these algorithms on them (the ones you suggested above).

**Comment by Korra15** [comment]
okay

**Comment by czgdp1807** [comment]
We should add a new data structure, `ImplicitGraph`. (Read #224).

**Comment by Korra15** [comment]
okay

**Comment by Abhinavcode13** [comment]
please assign me this issue @Korra15 , would love to resolve this issue under gssoc 23.

**Comment by Korra15** [comment]
@Abhinavcode13 you can contact the maintainers of this repo regarding the same
